utcp.c

```diff
@@ -1594,6 +1594,29 @@ int utcp_shutdown(struct utcp_connection *c, int dir) {
 	return 0;
 }
 
+// Closes all the opened connections
+int utcp_abort_all_connections(struct utcp *utcp) {
+	if(!utcp) {
+		return -1;
+	}
+
+	int total_connections = utcp->nconnections;
+
+	for(int i = 0; i < total_connections; i++) {
+		struct utcp_connection *c = utcp->connections[i];
+
+		if(c->recv) {
+      errno = 0;
+			c->recv(c, NULL, 0);
+    }
+
+		if(utcp_abort(c) != -1) {
+      utcp->nconnections = utcp->nconnections - 1;
+		}
+	}
+  return utcp->nconnections;
+}
+
```

Comment:
I would make this function just return `void`. Either a connection is aborted, or it wasn't in a valid state to begin with, so the end result is the same for all connections.

Reply:
How can I assert that all connections are closed or not by `utcp_abort()` ?

Reply:
The purpose of `utcp_abort()` is to close a connection and to drop any unacknowledged data, and send a RST packet to the peer to indicate this connection is aborted. So it has most of the semantics of `utcp_close()`, including invalidating the `utcp_connection_t` handle.

For the purpose of informing a blacklisted node that its channels are closed, we want to send the RST packet and drop any unacknowledged data, but we want to keep the `utcp_connection_t` handles valid, because the application still has `meshlink_channel_t` handles that it hasn't explicitly closed itself. So `utcp_abort_all_connections()` should not close any handles, they should be kept open. The receive callback will be called for each connection, so the application can respond accordingly.

Also, if `utcp_abort(c)` return `-1`, it means that either `c` is an invalid handle, or that the connection was already closed.